File: output/ChatGPT/codenet/Java/C++/s987117545.cpp
#include <iostream>

using namespace std;

int main() {
    long long n, k;
    const long long DEVISOR = 1000000000+7;

    cin >> n >> k;

    long long ans = 0;
    for (long long i = k; i <= n+1; i++) {
        long long max = i*(2*n-i+1)/2 ;
        long long min = i*(i-1)/2;
        ans += (max - min + 1) % DEVISOR;
    }
    cout << ans % DEVISOR << endl;

    return 0;
}

File: output/CodeGen/codenet/Java/Python/s987117545.py
def main():
    n = int(input())
    k = int(input())
    ans = 0
    for i in range(k, n+1):
        max = i*(2*n-i+1)/2
        min = i*(i-1)/2
        ans += (max - min + 1) % DEVISOR
    print(ans % DEVISOR)

if __name__ == '__main__':
    main()